plugins/basic/nexus-repository-docker/src/main/java/org/sonatype/nexus/repository/docker/internal/DockerFacetImpl.java

```diff
@@ -0,0 +1,682 @@
+    try {
+      return mapper.readValue(layerMetadataContent.get(), typeRef);
+    }
+    catch (Exception e) {
+      log.debug("Invalid JSON", e);
+      throw new V1Exception.InvalidJson();
+    }
+  }
+
+  /**
+   * Validate that images index is a proper JSON list containing layer references.
+   *
+   * @param imagesIndexContent images index content
+   * @return parsed images index
+   */
+  private List<Map<String, Object>> readImagesIndex(final Supplier<InputStream> imagesIndexContent) throws IOException {
+    try (InputStream in = imagesIndexContent.get()) {
+      TypeReference<List<Map<String, Object>>> typeRef = new TypeReference<List<Map<String, Object>>>() {}; //CHECKSTYLE:OFF
+      try {
+        return mapper.readValue(in, typeRef);
+      }
+      catch (Exception e) {
+        log.debug("Invalid JSON", e);
+        throw new V1Exception.InvalidJson();
+      }
+    }
+  }
+
+  /**
+   * Read layer id.
+   *
+   * @param streamSupplier layer id content
+   * @return parsed layer id
+   */
+  private String readLayerId(final Supplier<InputStream> streamSupplier) {
+    try {
+      return validateLayerId(mapper.readValue(streamSupplier.get(), String.class));
+    }
+    catch (Exception e) {
+      log.debug("Invalid JSON", e);
+      throw new V1Exception.InvalidJson("Invalid data");
+    }
+  }
+
+  /**
+   * Validate layer id.
+   *
+   * @param layerId to validate
+   * @return validated layer id
+   */
+  private String validateLayerId(final String layerId) {
+    if (!validLayerId.matcher(layerId).matches()) {
+      throw new V1Exception.InvalidImageId();
+    }
+    return layerId;
+  }
+
+  /**
+   * Validate tag name.
+   *
+   * @param tag to validate
+   */
+  private void validateTagName(final String tag) {
+    if (!validTag.matcher(tag).matches()) {
+      throw new V1Exception.InvalidTagName();
+    }
+  }
+
+  /**
+   * Convert an asset blob to {@link Content}.
+   *
+   * @param asset asset
+   * @param blob  blob
+   * @return content
+   */
+  private Content marshall(final Asset asset, final Blob blob) {
+    Content content = new Content(new BlobPayload(blob, asset.requireContentType()));
+    Content.extractFromAsset(asset, hashAlgorithms, content.getAttributes());
+    return content;
+  }
+
+  /**
+   * Create a supplier for an asset blob.
+   *
+   * @param tx    storage transaction
+   * @param asset asset
+   * @return asset blob content supplier
+   */
+  private Supplier<InputStream> blobStreamSupplier(final StorageTx tx, final Asset asset) {
+    return new Supplier<InputStream>()
+    {
+      @Override
+      public InputStream get() {
+        return tx.requireBlob(asset.requireBlobRef()).getInputStream();
+      }
+    };
+  }
+
+  /**
+   * Get the name of layer metadata asset.
+   *
+   * @param layerId layer id
+   * @return name
+   */
+  private String layerMetadataName(final String layerId) {return layerId + "/json";}
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move this closing curly brace to the next line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)